src/p2pMessage.ts

```diff
@@ -0,0 +1,300 @@
+// Copyright 2018 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import { NodeId } from "./sessionMessage";
+
+const RLP = require("rlp");
+
+export enum MessageType {
+    SYNC_ID = 0x00,
+    ACK_ID,
+    REQUEST_ID,
+    ALLOWED_ID,
+    DENIED_ID,
+    ENCRYPTED_ID,
+    UNENCRYPTED_ID
+}
+
+export class HandshakeMessage {
+    private body: SyncMessage | AckMessage;
+
+    constructor(body: SyncMessage | AckMessage) {
+        this.body = body;
+    }
+
+    toEncodeObject(): Array<any> | number | string {
```

Comment:
When does it return `number` or `string`?

Reply:
I removed it.